### PR TITLE
helm test - show log of test pod

### LINKED ISSUE #1957: add --verbose flag to helm test
When set to true, this flag will stream the test pod logs in the output

cc/ As per our discussion @foxish @erictune @vdice

### PR DESCRIPTION
**Pull request by GaoLiping**
fix(helm): add log of test pod to 'helm test'

When 'helm test' finishes, this will print the log of the test pods

Closes #1957

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -686,6 +686,37 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
 	return err
 }
 
+//GetPodLogs print the log of a pod to standard output
```
Comment (sorenmat) [inline pkg/kube/client.go]: Doesn't it just return the log?

File: pkg/kube/client.go [hunk 2]
```diff
@@ -686,6 +686,37 @@ func (c *Client) watchPodUntilComplete(timeout time.Duration, info *resource.Inf
...
+func (c *Client) GetPodLogs(namespace string, reader io.Reader, timeout time.Duration) (string, error) {
```
Comment (sorenmat) [inline pkg/kube/client.go]: doesn't seem like the timeout parameter is used

File: pkg/releasetesting/environment.go [hunk 3]
```diff
@@ -62,14 +62,24 @@ func (env *Environment) getTestPodStatus(test *test) (core.PodPhase, error) {
...
 }
 
+func (env *Environment) getTestPodsLogs(test *test) (string, error) {
+	b := bytes.NewBufferString(test.manifest)
+	log, err := env.KubeClient.GetPodLogs(env.Namespace, b, time.Duration(env.Timeout)*time.Second)
+	if err != nil {
+		return "", err
+	}
+
+	return log, err
```
Comment (sorenmat) [inline pkg/releasetesting/environment.go]: I personally like to explicitly return nil to indicate that there were no error

File: pkg/releasetesting/test_suite.go [hunk 4]
```diff
@@ -92,6 +92,8 @@ func (ts *TestSuite) Run(env *Environment) error {
 		resourceCleanExit := true
 		status := core.PodUnknown
 		if resourceCreated {
+			fmt.Println("testing pod created ")
```
Comment (sorenmat) [inline pkg/releasetesting/test_suite.go]: remove this perhaps?

**Comment by frankh** [comment]
Hi, I didn't realise you already had a PR for this,

I've made one which does a similar thing, https://github.com/kubernetes/helm/pull/4081

The main differences are: It adds a flag `--logs` for showing the logs, and it streams the logs in realtime instead of printing them all at once

I would appreciate your input and which approach you think works best

**Review by michelleN — CHANGES_REQUESTED** [review]
Hey @GaoLiping! Thanks for contributing. It looks like this pull request was made against the `release-2.8` branch. Would you mind making this pull request against `master`? Right now, I can't test it manually because there are merge conflicts against master. I'm happy to pair and help you on that if you need. 

@frankh - Thanks for touching base here to coordinate on similar functionality. It looks like you closed the `--logs` flag PR to focus on `--paralllel`. I think the streaming functionality is what folks have been asking for as well which is what makes sense to me as well.

I think doing a `--logs` or `--verbose` flag makes sense. If we put this functionality behind a flag called `--dump-logs` then we can have both play nice but still keep the simplified output we have already.

Let me know what ya'll think @GaoLiping @frankh

**Comment by michelleN** [comment]
@GaoLiping are you still interested in contributing this feature?

**Comment by jeff-knurek** [comment]
I didn't see this PR till I already finished most of the work to do similar, but as it's been ignored for some time, maybe someone can have a look at https://github.com/helm/helm/pull/6167

my take is that `--verbose` would be the option that would stream logs in realtime (like what was started in https://github.com/helm/helm/pull/4081) whereas `--logs` is suitable for getting a dump of the logs at the end. and i suppose that either option will be satisfactory till Helm 3 changes are available https://github.com/helm/helm/issues/6020

**Comment by bacongobbler** [comment]
Closing this PR due to inactivity. If someone is interested in pushing this along the finish line, please write a PR against Helm 3. As per [the announcement on the mailing list](https://lists.cncf.io/g/cncf-helm/topic/last_v2_feature_release/34301238), Helm 2.15.0 is the last release where we will be merging new features for Helm 2. Thanks!

**Pull request closed without merging**